docs/core/v3.0.rst

```diff
@@ -1042,11 +1079,14 @@ The following names are optional:
     compressor is used.
 
 ``storage_transformers``
+^^^^^^^^^^^^^^^^^^^^^^^^
 
     Specifies a stack of `storage transformers`_. Each value in the list must
-    be an object containing the name ``storage_transformer`` whose value
-    is a URI that identifies a storage transformer and dereferences to a
-    human-readable representation of the codec specification. The
+    be an object containing the names ``extension`` and ``type``.
```

Comment:
I like the use of `type` rather than `storage_transformers` as the identifier key.

However, what is the reason to have both `extension` and `type`?  Is the idea that a single "extension" may define more than one type?

I think it would be simpler to just have a single key, `type`, which is the URI.  If a single extension defines more than one type, the URI could have a fragment identifier, like: `https://whatever/path#some-type`

Reply:
The reason was to use the same structure `data_type`, `chunk_grid` and `chunk_memory_layout` already have. I think using the same structure across different metadata objects helps for consistency, but I agree that there might be a more elegant solution. What about just using `extension` as the URI key, and having `format` back in the `configuration` object?

Reply:
Thanks for the clarification, I missed the fact the extension/type structure was already being used.  Maybe for uniformity all of these extension points should just be identified by a url unconditionally even if the "type" is defined by the core specification.

E.g.

```json
{"chunk_layout": {"type": "https://zarr.dev/...#regular_grid", "configuration": ...},
 "data_type": {"type": "https://zarr.dev/...#float32"},
 "compression": {"type": "https://zarr.dev/...#zstd"}
}
```

I think putting the extension urls in a separate "extensions" property is more complicated and also would create the possibility of a conflict if the user wishes to use at the same time two extensions that happen to use the same "type" value.

Reply:
I'll leave it as-is for now, let's discuss this separately.